src/analyzer/plugins/analyzerqueenmarybeats.cpp

```diff
@@ -70,47 +72,28 @@ bool AnalyzerQueenMaryBeats::finalize() {
     }
 
     std::vector<double> df;
-    std::vector<double> beatPeriod(nonZeroCount);
+    std::vector<double> beatPeriod;
     std::vector<double> tempi;
+    const auto required_size = std::max(0, nonZeroCount - 2);
+    df.reserve(required_size);
+    beatPeriod.reserve(required_size);
 
-    df.reserve(nonZeroCount);
-
-    for (int i = 0; i < nonZeroCount; ++i) {
+    // skip first 2 results as it might have detect noise as onset
+    // that's how vamp does and seems works best this way
+    for (int i = 2; i < nonZeroCount; ++i) {
         df.push_back(m_detectionResults.at(i));
+        beatPeriod.push_back(0.0);
     }
 
-    TempoTrackV2 tt(m_iSampleRate, kStepSize);
+    TempoTrackV2 tt(m_iSampleRate, m_stepSize);
     tt.calculateBeatPeriod(df, beatPeriod, tempi);
 
     std::vector<double> beats;
     tt.calculateBeats(df, beatPeriod, beats);
 
-    // In some tracks a beat at 0:00 is detected when a noise floor starts.
-    // Here we check the level and the position for plausibility and remove
-    // the beat if this is the case.
-    size_t firstBeat = 0;
-    if (beats.size() >= 3) {
-        if (beats.at(0) <= 0) {
-            firstBeat = 1;
-        } else if (m_detectionResults.at(beats.at(0)) <
-                (m_detectionResults.at(beats.at(1)) +
-                m_detectionResults.at(beats.at(2))) / 4) {
-            // the beat is not half es high than the average of the two
-            // following beats. Skip it.
-            firstBeat = 1;
-        } else {
-            int diff = (beats.at(1) - beats.at(0)) - (beats.at(2) - beats.at(1));
-            // we don't allow a significant tempo change after the first beat
-            if (diff > 2 || diff < -2) {
-                // first beat is off grid. Skip it.
-                firstBeat = 1;
-            }
-        }
-    }
-
     m_resultBeats.reserve(beats.size());
-    for (size_t i = firstBeat; i < beats.size(); ++i) {
-        double result = (beats.at(i) * kStepSize) - kStepSize / 2;
+    for (size_t i = 0; i < beats.size(); ++i) {
```

Comment:
Why not use a range-based for loop?

Reply:
This is not new code, just changed the loop start, should rewrite it range-based for loop?

Reply:
Then let's keep it as is.